### ISSUE TITLE
USB screen turns on and mirror correctly but display is frozen

### ISSUE THREAD

**Issue by aleph-zer0** [DisplayLink/evdi#68]
Hi 

I have been installing evdi on Debian stretch using the script provided by adnanhozdic here:
https://github.com/AdnanHodzic/displaylink-debian/

Everything went smooth, but when dlm.service and turning on my USB monitor, it just freezes. This is a duplicate of the issue I posted here: 
https://github.com/AdnanHodzic/displaylink-debian/issues/47

A similar issue has been documented there in the comments, where using side by side instead of mirror only display the mouse pointer on either a desktop with no windows or black background. 

Any help would be much appreciated. 

So dlm.service runs correctly on debian Jessie stretch with kernel 3.16.0-4. The USB monitor I am trying to set-up is a AOC E1759FWU and computer is an Intel NUC i7 with Iris 6100 graphic.

xrandr sees both monitors within providers and I can access config for both through xfce display GUI.
Upon running:
`xrandr --setprovideroutputsource 1 0`
the usb monitor successfully turns on and mirrors main screen but freeze and does not refresh afterward (like displaying a print screen of when it was connected).

Tried so far:
-connecting on USB 2 or 3 with or without second power USB
-update/upgrade and reboot
-rerun display Link script / uninstall and reinstall successfully
-updated intel graphic module
-restart dlm.service (update print screen but still no refresh)

sample output below

```
x@cube ~ $ uname -a
Linux cube 3.16.0-4-amd64 #1 SMP Debian 3.16.36-1+deb8u2 (2016-10-19) x86_64 GNU/Linux

x@cube ~ $ xrandr --listproviders
Providers: number : 2
Provider 0: id: 0x45 cap: 0xf, Source Output, Sink Output, Source Offload, Sink Offload crtcs: 3 outputs: 3 associated providers: 1 name:modesetting
Provider 1: id: 0x125 cap: 0x2, Sink Output crtcs: 1 outputs: 1 associated providers: 1 name:modesetting

x@cube ~ $ sudo systemctl status dlm.service
● dlm.service - DisplayLink Manager Service
Loaded: loaded (/lib/systemd/system/dlm.service; enabled; vendor preset: enabled)
Active: active (running) since Thu 2016-12-29 23:16:46 GMT; 5min ago
Process: 3710 ExecStartPre=/sbin/modprobe evdi (code=exited, status=0/SUCCESS)
Main PID: 3714 (DisplayLinkMana)
CGroup: /system.slice/dlm.service
└─3714 /opt/displaylink/DisplayLinkManager

Dec 29 23:16:46 cube systemd[1]: Starting DisplayLink Manager Service...
Dec 29 23:16:46 cube systemd[1]: Started DisplayLink Manager Service.


```

**Comment by tristan3214** [comment]
I want to add to this. I recently purchased an Asus MB16b and have this same issue. I am running xubuntu 16.04 LTS with kernel 4.4.  The displaylink driver seems to install just fine (I can find Displaylink manager in the running processes and the monitor is detected by the xfce display manager. Going further Xrandr detects the display and I get the same issue when using Xrandr on the command line.  To reiterate, when I try to switch to using only the USB display it just shows a frozen screen and then goes black after 5 or so seconds. 

I have essentially done all that this fellow has, as well,  to try to resolve the issue. Any assistance would be wonderful.

**Comment by csullivannet** [comment]
Has anyone had any success with this issue? I'm running Fedora 24 on a Lenovo Y700-15ISK.

**Comment by aleph-zer0** [comment]
> Has anyone had any success with this issue? I'm running Fedora 24 on a Lenovo Y700-15ISK.

hi I did not really had time to try more stuff lately but I will give it another go in the coming weeks. Will keep this thread  updated with further progresses

**Comment by nexusVI** [comment]
I am seeing the same problem.

* Debian 9 (Stretch)
* i3wm
* compton

Tried adding a modeline manually as suggested here:
https://wiki.archlinux.org/index.php/DisplayLink#Screen_redraw_is_broken

No success. The same screen used to work until I upgraded to kernel 4.8, iirc - though it had many flaws, like flickering, slow redraw, ... I sort of expected progress regarding this issue with the new release, but now it's worse than ever :-(

**Comment by aiguofer** [comment]
I've had this issue ever since I upgraded to Fedora 25 I think (maybe even earlier)... I'm on fedora 26 alpha now and tried the latest devel (since the latest stable doesn't compile) and it's still the same issue. I know they'll say "the latest devel might not be compatible with current displaymanager" but for a few months now my dock has been useless.

**Comment by vnenkpet** [comment]
I am having the exact same problem on Ubuntu 17, worked before with Ubuntu 14. Can't find a solution.

**Comment by dalangston** [comment]
I was having the same problem with Debian 9.  I found this solution on the [DisplayLink Forum](http://displaylink.org/forum/showthread.php?s=a44a80a58d5dcd20d36b324bb1efc976&t=65258).  

Create the file ``` /etc/X11/xorg.conf.d/20-intel.conf``` with this content
```
Section "Device"
Identifier "Intel Graphics"
Driver "Intel"
EndSection
```

After a restart my displays are working correctly.

**Comment by aiguofer** [comment]
Omg!! thank you @dalangston that worked for me too!! my dock is no longer just a USB hub! :)

**Comment by aremai** [comment]
i've tried that...but it didn't change anything for me.
the behavior is still the same. it's stilly laggy and unusable for me :(

my kernel is: 4.4.0-78-generic
and i'm using kde plasma 5.8.6

**Comment by kamilla** [comment]
Any updates on the issue? I just upgraded my Debian Jessie to Stretch and now I have the same problem. Jessie was ok, just few glitches and connection problems sometimes, but now the external dislpay is unusable.

The display just freezes on connection and displays the same output as one of the other displays even if it is not mirrored (Sometimes theres only black screen though freezed). No action is drawn to screen but if I have two other displays connected (laptop and laptop HDMI) I can move my mouse from another display to another and the mouse speed and position is maintained correctly in that freezed/black display in the middle but not drawn.

I've tried all the suggested (and not suggested) methods from reinstalling displaylink driver to connecting via different ports, with or without other displays, playing with dlm.service or forcing certaing configurations with no success.

I have a feeling that it could have something to do with USB, maybe hardware or driver problem. Not with the displaylink driver itself and not with the USB replicator. But I haven't had any luck so far.

I'm using HP Pavilion Notebook (Z5F67EA#UUW) with HP 3005pr USB 3.0 port replicator. External displays are Acer R231 23" monitors.

At least if nothing else works, I can always downgrade back to Debian Jessie, but would be uncool now that Stretch was released.

`λ ~/ sudo dmesg
..
[83770.315783] usb 1-1: new high-speed USB device number 53 using xhci_hcd
[83770.460404] usb 1-1: New USB device found, idVendor=0451, idProduct=8042
[83770.460416] usb 1-1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[83770.461623] hub 1-1:1.0: USB hub found
[83770.461679] hub 1-1:1.0: 4 ports detected
[83770.579713] usb 2-1: new SuperSpeed USB device number 18 using xhci_hcd
[83770.599900] usb 2-1: New USB device found, idVendor=0451, idProduct=8040
[83770.599903] usb 2-1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[83770.600604] hub 2-1:1.0: USB hub found
[83770.600622] hub 2-1:1.0: 4 ports detected
[83770.747671] usb 1-1.3: new high-speed USB device number 54 using xhci_hcd
[83770.847894] usb 1-1.3: New USB device found, idVendor=0424, idProduct=2517
[83770.847903] usb 1-1.3: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[83770.848672] hub 1-1.3:1.0: USB hub found
[83770.848711] hub 1-1.3:1.0: 7 ports detected
[83771.139983] usb 1-1.3.1: new high-speed USB device number 55 using xhci_hcd
[83771.250469] usb 1-1.3.1: New USB device found, idVendor=048d, idProduct=1165
[83771.250473] usb 1-1.3.1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[83771.250475] usb 1-1.3.1: Product: USB Mass Storage Device
[83771.250476] usb 1-1.3.1: Manufacturer: iTE Tech
[83771.250478] usb 1-1.3.1: SerialNumber: 0000000000005131
[83771.251141] usb-storage 1-1.3.1:1.0: USB Mass Storage device detected
[83771.251244] scsi host3: usb-storage 1-1.3.1:1.0
[83772.083853] usb 2-1.1: new SuperSpeed USB device number 19 using xhci_hcd
[83772.104455] usb 2-1.1: New USB device found, idVendor=17e9, idProduct=430a
[83772.104467] usb 2-1.1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[83772.104472] usb 2-1.1: Product: HP USB 3.0 Port Replicator
[83772.104476] usb 2-1.1: Manufacturer: DisplayLink
[83772.104480] usb 2-1.1: SerialNumber: 0028104
[83772.114509] usb 2-1.1: Warning! Unlikely big volume range (=511), cval->res is probably wrong.
[83772.114525] usb 2-1.1: [15] FU [HP USB Media Audio Playback Volume] ch = 6, val = -8176/0/16
[83772.115137] usb 2-1.1: Warning! Unlikely big volume range (=767), cval->res is probably wrong.
[83772.115143] usb 2-1.1: [12] FU [Mic Capture Volume] ch = 2, val = -4592/7680/16
[83772.140049] cdc_ncm 2-1.1:1.5: MAC-Address: 00:1f:b5:22:94:19
[83772.140059] cdc_ncm 2-1.1:1.5: setting rx_max = 16384
[83772.140226] cdc_ncm 2-1.1:1.5: setting tx_max = 16384
[83772.140784] cdc_ncm 2-1.1:1.5 usb0: register 'cdc_ncm' at usb-0000:00:14.0-1.1, CDC NCM, 00:1f:b5:22:94:19
[83772.205657] cdc_ncm 2-1.1:1.5 enx001fb5229419: renamed from usb0
[83772.242393] IPv6: ADDRCONF(NETDEV_UP): enx001fb5229419: link is not ready
[83772.242475] IPv6: ADDRCONF(NETDEV_UP): enx001fb5229419: link is not ready
[83772.284388] scsi 3:0:0:0: Direct-Access     IT1165   USB Flash Disk   0.00 PQ: 0 ANSI: 4
[83772.284645] sd 3:0:0:0: Attached scsi generic sg1 type 0
[83772.285772] sd 3:0:0:0: [sdb] 246784 512-byte logical blocks: (126 MB/121 MiB)
[83772.286526] sd 3:0:0:0: [sdb] Write Protect is on
[83772.286530] sd 3:0:0:0: [sdb] Mode Sense: 0b 00 80 00
[83772.287313] sd 3:0:0:0: [sdb] No Caching mode page found
[83772.287317] sd 3:0:0:0: [sdb] Assuming drive cache: write through
[83772.292858]  sdb: sdb1
[83772.299766] sd 3:0:0:0: [sdb] Attached SCSI removable disk
[83772.723489] evdi: [W] evdi_painter_disconnect:462 (dev=-1) An unknown connection to ffff976dfb37d400 tries to close us
[83772.723492] evdi: [W] evdi_painter_disconnect:463  - ignoring
[83772.729003] evdi: [D] evdi_painter_connect:443 (dev=1) Connected with ffff976dfb37d400
[83772.729006] evdi: [D] evdi_detect:75 (dev=1) Painter is connected
[83772.729069] evdi: [D] evdi_painter_crtc_state_notify:360 (dev=1) Notifying crtc state: 3
[83772.729920] evdi: [D] evdi_detect:75 (dev=1) Painter is connected
[83772.729925] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[83772.812695] evdi: [D] evdi_detect:75 (dev=1) Painter is connected
[83772.812699] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[83772.894681] evdi: [D] evdi_detect:75 (dev=1) Painter is connected
[83772.894692] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[83772.924982] evdi: [D] evdi_painter_mode_changed_notify:376 (dev=1) Notifying mode changed: 1920x1080@60; bpp 32; 
[83772.924986] evdi: [D] evdi_painter_mode_changed_notify:377 pixel format 875713112
[83772.925021] evdi: [D] evdi_painter_crtc_state_notify:360 (dev=1) Notifying crtc state: 0
[83772.925025] evdi: [D] evdi_painter_dpms_notify:347 (dev=1) Notifying dpms mode: 0
[83773.033884] evdi: [D] evdi_detect:75 (dev=1) Painter is connected
[83773.033887] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[83773.088265] evdi: [D] evdi_detect:75 (dev=1) Painter is connected
[83773.088269] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[83773.175601] evdi: [D] evdi_detect:75 (dev=1) Painter is connected
[83773.175608] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[83773.261820] evdi: [D] evdi_detect:75 (dev=1) Painter is connected
[83773.261823] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[83773.324463] evdi: [D] evdi_detect:75 (dev=1) Painter is connected
[83773.324468] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[83773.381581] evdi: [D] evdi_detect:75 (dev=1) Painter is connected
[83773.381585] evdi: [D] evdi_painter_get_edid_copy:192 (dev=1) 00 ff ff
[83775.256560] cdc_ncm 2-1.1:1.5 enx001fb5229419: network connection: disconnected
`

`
λ ~/ sudo lshw
    ..
     *-pci                                                                                                                                                                                                                                     
          description: Host bridge                 
          product: Intel Corporation            
          vendor: Intel Corporation           
          physical id: 100                                                        
          bus info: pci@0000:00:00.0                      
          version: 02                                                            
          width: 32 bits                
          clock: 33MHz                                      
        *-display                                                                 
             description: VGA compatible controller                                          
             product: Intel Corporation                                          
             vendor: Intel Corporation                                                                                                                                                                                                 
             physical id: 2                          
             bus info: pci@0000:00:02.0            
             version: 02                                                
             width: 64 bits                          
             clock: 33MHz                                        
             capabilities: pciexpress msi pm vga_controller bus_master cap_list rom
             configuration: driver=i915 latency=0
             resources: irq:132 memory:b0000000-b0ffffff memory:a0000000-afffffff ioport:4000(size=64) memory:c0000-dffff                    
        *-generic:0                                                                                                                                                                                         
             description: Signal processing controller                                                    
             product: Skylake Processor Thermal Subsystem         
             vendor: Intel Corporation                                                                                                                                                                                               
             physical id: 4                                    
             bus info: pci@0000:00:04.0                                         
             version: 02                                                                                                                                                                                                                                                              
             width: 64 bits                          
             clock: 33MHz                          
             capabilities: msi pm cap_list                  
             configuration: driver=proc_thermal latency=0
             resources: irq:16 memory:b1320000-b1327fff                           
        *-usb                                                     
             description: USB controller                
             product: Sunrise Point-LP USB 3.0 xHCI Controller                                         
             vendor: Intel Corporation                                     
             physical id: 14                                                                                             
             bus info: pci@0000:00:14.0
             version: 21                       
             width: 64 bits                                           
             clock: 33MHz                        
             capabilities: pm msi xhci bus_master cap_list                                                                                                             
             configuration: driver=xhci_hcd latency=0
             resources: irq:127 memory:b1300000-b130ffff                          
           *-usbhost:0                                                          
                product: xHCI Host Controller           
                vendor: Linux 4.9.0-3-amd64 xhci-hcd
                physical id: 0                 
                bus info: usb@1                   
                logical name: usb1                                                                                                                                                                                                
                version: 4.09       
                capabilities: usb-2.00                                            
                configuration: driver=hub slots=12 speed=480Mbit/s
              *-usb:0                                                                                        
                   description: USB hub                                     
                   product: Hub        
                   vendor: Texas Instruments, Inc.             
                   physical id: 1                                                             
                   bus info: usb@1:1                                                                                                                                              
                   version: 1.00                                
                   capabilities: usb-2.10
                   configuration: driver=hub slots=4 speed=480Mbit/s
                 *-usb
                      description: USB hub
                      product: Hub
                      vendor: Standard Microsystems Corp.
                      physical id: 3
                      bus info: usb@1:1.3
                      version: 0.02
                      capabilities: usb-2.00
                      configuration: driver=hub maxpower=2mA slots=7 speed=480Mbit/s
                    *-usb
                         description: Mass storage device
                         product: USB Mass Storage Device
                         vendor: iTE Tech
                         physical id: 1
                         bus info: usb@1:1.3.1
                         logical name: scsi3
                         version: 1.00
                         serial: 0000000000005131
                         capabilities: usb-2.00 scsi emulated
                         configuration: driver=usb-storage maxpower=480mA speed=480Mbit/s
                       *-disk
                            description: SCSI Disk
                            product: USB Flash Disk
                            vendor: IT1165
                            physical id: 0.0.0
                            bus info: scsi@3:0.0.0
                            logical name: /dev/sdb
                            version: 0.00
                            serial: 1165
                            size: 120MiB (126MB)
                            capabilities: removable
                            configuration: ansiversion=4 logicalsectorsize=512 sectorsize=512
                          *-medium
                               physical id: 0
                               logical name: /dev/sdb
                               size: 120MiB (126MB)
                               capabilities: partitioned partitioned:dos
                             *-volume
                                  description: Windows FAT volume
                                  vendor: MSWIN4.1
                                  physical id: 1
                                  logical name: /dev/sdb1
                                  logical name: /media/kamilla/HP Dock
                                  version: FAT16
                                  serial: 0000-0001
                                  size: 120MiB
                                  capacity: 120MiB
                                  capabilities: primary bootable fat initialized
                                  configuration: FATs=2 filesystem=fat label=HP Dock mount.fstype=vfat mount.options=ro,nosuid,nodev,relatime,uid=1000,gid=1000,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,showexec,utf8,flush,errors=remount-ro state=mounted
              *-usb:1                                
                   description: Keyboard           
                   product: USB Keyboard                
                   vendor: Logitech
                   physical id: 2        
                   bus info: usb@1:2                              
                   version: 64.00     
                   capabilities: usb-1.10
                   configuration: driver=usbhid maxpower=90mA speed=2Mbit/s
              *-usb:2   
                   description: Video
                   product: HP Wide Vision HD
                   vendor: DFKAR01BI41E4T                             
                   physical id: 3                
                   bus info: usb@1:3                                                                                                                                   
                   version: 0.04
                   capabilities: usb-2.00
                   configuration: driver=uvcvideo maxpower=500mA speed=480Mbit/s
              *-usb:3                 
                   description: Mouse
                   product: G9 Laser Mouse
                   vendor: Logitech
                   physical id: 6
                   bus info: usb@1:6
                   version: 50.00
                   serial: 19464B81D50029
                   capabilities: usb-2.00
                   configuration: driver=usbhid maxpower=98mA speed=12Mbit/s
              *-usb:4
                   description: Bluetooth wireless interface
                   vendor: Intel Corp.
                   physical id: 7
                   bus info: usb@1:7
                   version: 0.01
                   capabilities: bluetooth usb-2.00
                   configuration: driver=btusb maxpower=100mA speed=12Mbit/s
           *-usbhost:1
                product: xHCI Host Controller
                vendor: Linux 4.9.0-3-amd64 xhci-hcd
                physical id: 1
                bus info: usb@2
                logical name: usb2
                version: 4.09
                capabilities: usb-3.00
                configuration: driver=hub slots=6 speed=5000Mbit/s
              *-usb
                   description: USB hub
                   vendor: Texas Instruments, Inc.
                   physical id: 1
                   bus info: usb@2:1
                   version: 1.00
                   capabilities: usb-3.00
                   configuration: driver=hub slots=4 speed=5000Mbit/s
                 *-usb
                      description: Audio device
                      product: HP USB 3.0 Port Replicator
                      vendor: DisplayLink
                      physical id: 1
                      bus info: usb@2:1.1
                      version: 1.05
                      serial: 0028104
                      capabilities: usb-3.10 audio-control
                      configuration: driver=cdc_ncm maxpower=8mA speed=5000Mbit/s
    ..
`

`
λ ~/ xrandr                              
Screen 0: minimum 320 x 200, current 3840 x 1080, maximum 8192 x 8192          
eDP-1 connected 1920x1080+1920+0 (normal left inverted right x axis y axis) 308mm x 173mm
   1920x1080     60.01*+  59.93    40.00 
   1680x1050     59.95    59.88                                                
   ..                                             
HDMI-1 connected 1920x1080+0+0 (normal left inverted right x axis y axis) 509mm x 286mm
   1920x1080     60.00*+  50.00    59.94                                       
   1920x1080i    60.00    50.00    59.94                                      
   ..                                           
DVI-I-1-1 connected 1920x1080+0+0 (normal left inverted right x axis y axis) 509mm x 286mm
   1920x1080     60.00*+  50.00    59.94    60.01                              
   1680x1050     59.88                                                        
   ..
`

**Comment by dalangston** [comment]
@kamilla I see that `xrandr --listproviders` shows two entries as `modesetting`.  It seems that sometimes the system does not correctly select the correct driver for each output source.  I had the same problem and fixed it by adding the following to `/etc/X11/xorg.conf.d/20-intel.conf` :
```
Section "Device"
Identifier "Intel Graphics"
Driver "Intel"
EndSection
```
After a reboot my displays worked properly.

**Comment by aiguofer** [comment]
I don't use Debian, so I'm not sure what DE they use by default, but did they by chance switch from Xorg to Wayland in that upgrade? that might be something to look at.

One that that happens to me sometimes (if I boot while plugged in) is that GDM seems to take over the external monitors, so when I log in, the external monitors are actually extending GDM and not my user session. I usually boot first, log in, and then plug it in.

What did the trick for me was ensuring `/etc/X11/xorg.conf.d/20-intel.conf` had the above mentioned contents and making sure that both GDM and session were started using Xorg instead of Wayland.

**Comment by kamilla** [comment]
@dalangston, thank you for your reply. I noticed your suggestion earlier, but didn't think much of it since Jessie worked nicely with probing hardware on start (without xorg.conf), so I thought it wouldn't matter. But now that I added xorg.conf it works! Some mouse cursor flickering, but I need to investigate that later on.

@aiguofer, no, Debian doesn't use Wayland by default, though it can be used if wanted. I use X (xfce4 + lightdm). But thank you for your information. It can be useful to someone else.

Anyways, it works now, so thank you for your help. And if someone else struggles with the same problem on Debian, I can suggest to run X -configure / manually add/edit xorg.conf to make things run nicely again.
